src/routes/UserSettings/UserSettings.connector.js

```diff
@@ -63,21 +69,36 @@ export function mapStateToProps (state, props) {
     fetchPending,
     allCommunitiesSettings,
     messageSettings,
+    searches,
     queryParams
   }
 }
 
-export const mapDispatchToProps = {
-  fetchForCurrentUser,
-  updateUserSettings,
-  unBlockUser,
-  leaveCommunity,
-  loginWithService,
-  unlinkAccount,
-  setConfirmBeforeClose,
-  updateMembershipSettings,
-  updateAllMemberships,
-  registerStripeAccount
+export function mapDispatchToProps (dispatch) {
+  return {
+    fetchForCurrentUser: (params) => dispatch(fetchForCurrentUser(params)),
+    fetchSavedSearches: (params) => dispatch(fetchSavedSearches(params)),
+    deleteSearch: (params) => dispatch(deleteSearch(params)),
+    updateUserSettings: (params) => dispatch(updateUserSettings(params)),
+    unBlockUser: (params) => dispatch(unBlockUser(params)),
+    leaveCommunity: (params) => dispatch(leaveCommunity(params)),
+    loginWithService: (params) => dispatch(loginWithService(params)),
+    unlinkAccount: (params) => dispatch(unlinkAccount(params)),
+    setConfirmBeforeClose: (params) => dispatch(setConfirmBeforeClose(params)),
+    updateMembershipSettings: (params) => dispatch(updateMembershipSettings(params)),
+    updateAllMemberships: (params) => dispatch(updateAllMemberships(params)),
+    registerStripeAccount: (params) => dispatch(registerStripeAccount(params)),
+    viewSavedSearch: (params, path, selectedSearch) => {
+      const { boundingBox, featureTypes, search, topics } = params
+      dispatch(fetchMembers({ ...params }))
+      dispatch(fetchPosts({ ...params }))
+      dispatch(fetchPublicCommunities({ ...params }))
```

Comment:
again this all feels weird, why do we need this in the UserSettings?

Reply:
Was to provide the necessary updates to the store... but I've replaced that with [this](https://github.com/Hylozoic/hylo-evo/pull/663/commits/d76cbcc57adebd6875c90ba931d684890966c35d#diff-fbe00c3522dc75540d6eedef7bce90e7df0a0c88efd8db27db585ba75f19223bR111)